src/sage/rings/function_field/drinfeld_modules/drinfeld_module.py

```diff
@@ -1061,17 +1275,126 @@ def is_finite(self):
         from sage.rings.function_field.drinfeld_modules.finite_drinfeld_module import FiniteDrinfeldModule
         return isinstance(self, FiniteDrinfeldModule)
 
-    def j_invariant(self):
+    def is_isomorphic(self, psi):
         r"""
-        Return the j-invariant of the Drinfeld module if the rank is
-        two; raise a NotImplementedError otherwise.
+        Return ``True``  whether ``self`` is isomorphic to the other
+        Drinfeld module
 
-        Assume the rank is two. Write the generator `\phi_T = \omega +
-        g\tau + \Delta\tau^2`. The j-invariant is defined by
-        `\frac{g^{q+1}}{\Delta}`, `q` being the order of the base field
-        of the function ring. In our case, this field is always finite.
+        EXAMPLES::
 
-        OUTPUT: an element in the base codomain
+            sage: A = GF(5)['T']
+            sage: K.<T> = Frac(A)
+            sage: c = T^4 + 3*T^2 + T
+            sage: phi = DrinfeldModule(A, [T, T^3, T^9, T])
+            sage: psi = DrinfeldModule(A, [T, c^4*T^3, c^(24)*T^9, c^(124)*T])
+            sage: phi.is_isomorphic(psi)
+            True
+            sage: rho = DrinfeldModule(A, [T, 1, 1, 1])
+            sage: phi.is_isomorphic(rho)
+            False
+
+        ::
+
+            sage: phi = DrinfeldModule(A, [T, 1, 0, 1, 0, 0, 0, T^2, 1, 1])
+            sage: phi.is_isomorphic(phi)
+            True
+            sage: psi = DrinfeldModule(A, [T, 1, 0, 1, 0, T, 0, T^2, 1, 1])
+            sage: phi.is_isomorphic(psi)
+            False
+
+        TESTS::
+
+            sage: F.<a> = GF(5^2)
+            sage: A = F['T']
+            sage: phi = DrinfeldModule(A, [a, 1, 1])
+            sage: psi = DrinfeldModule(A, [Frac(A).gen(), 1, 1])
+            sage: phi.is_isomorphic(psi)
+            False
+        """
+        # Trivial checks:
+        if self.characteristic() != psi.characteristic():
+            return False
```

Comment:
I think actually that you should check that the categories are the same; indeed, you can have the same characteristic with different defining morphism $A \to K$.

When defining morphisms differ (but characteristic are equal), it is not clear to me what this function should do. I think that it's quite possible that the Drinfeld modules are isomorphic (e.g. take a Drinfeld module and twist it by an automorphism of $K$) but the isomorphism is then necessarily outside the category.
Probably, the good thing to do is just to raise an error when the categories are not the same.

Let me note furthermore that your function probably returns a wrong answer in this situation; indeed, I think that you should check that the $j$-invariants correspond one to the other under the automorphism $f$ of $K$ such that $f \circ \sigma_1 = \sigma_2$ (which is different from being equal).

Reply:
Good point, I added the category check. 

About checking that the $j$-invariants correspond one to the other under the automorphism $f$ of $K$, I'm not so sure to understand what you mean. I might have to check back, but I think the result of Potemine works for general $A$-fields.

Reply:
The question is: if you have a Drinfeld $\phi : A \to K\{\tau\}$ and an automorphism $f : K \to K$ and you define the Drinfeld module $\psi = f \circ \phi$ (where I extend $f$ to $K\{\tau\}$ by letting it act coefficientwise), should we consider that $\phi$ and $\psi$ are isomorphic?

I think it could make sense (although I prefer raising an error in this case).
However, the $j$-invariants are not the same in $K$: those of $\psi$ are the image by $f$ of those of $\phi$.

Reply:
For the category check, as I said, I would prefer raising an error if the categories differ.
However, if you insist on returning False, it would also be fine with me.

Reply:
> The question is: if you have a Drinfeld ϕ:A→Kτ and an automorphism f:K→K and you define the Drinfeld module ψ=f∘ϕ (where I extend f to Kτ by letting it act coefficientwise), should we consider that ϕ and ψ are isomorphic?
> 
> I think it could make sense (although I prefer raising an error in this case). However, the j-invariants are not the same in K: those of ψ are the image by f of those of ϕ.

I've given a bit of thoughts about this and I'm not sure if it is true in general that twists of Drinfeld module are isomorphic (which seems to be a crucial difference from the elliptic curve theory). The basic $j$-invariants test whether there exists an isomorphism $P\in F^{\times}$ such that $P\phi P^{-1} = \psi$, which is the same as saying that $g_i(\psi) = P^{1 - q^i} g_i(\phi)$ for all coefficients. If you twist a Drinfeld module and compare the coefficients, they will not satisfy this condition in general.